doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,132 @@
+
+Instead we recommend to directly bundle the module into your Lua scripts before you install them using [`amalg`](https://github.com/siffiejoe/lua-amalg/) (as [LuaRocks package](https://luarocks.org/modules/siffiejoe/amalg)). Check the [README](https://github.com/siffiejoe/lua-amalg/blob/master/README.md) for instructions on how to use `amalg`.
+
+In order to be able to use embedded packages on Exasol, you need to start your script with the following loader code:
+
+```lua
+table.insert(package.loaders,
+    function (module_name)
+        local loader = package.preload[module_name]
+        if not loader then
+            error("F-ML-1: Module " .. module_name .. " not found in package.preload.")
+        else
+            return loader
+        end
+    end
+)
+```
+
+## Using `exasollog`
+
+### Loading the Module
+
+The first thing you need is obviously a `require` instruction. In the simplest possible case, that is:
+
+```lua
+local log = require("exasollog")
+```
+
+### Configuring `exasollog`
+
+There are two ways to configure the module. One is regular configuration through setters.
+
+The other one is through an `init()` function. You will rarely ever need the second one. It is intended for one-time settings and mainly exists to make unit testing easier. You can control the timestamp pattern and use of the high-resolution timer through `init()`.
+
+While this is optional, we recommend using `set_client_name()` before your first log message, so that you get nice environment information at the beginning of each log. The client name should identify the application that uses the log and it's version number. But you could also write any other environment information into it that you want to appear right in front of the log.
+
+```lua
+log.set_client_name("my-script 1.3.0")
+```
+
+`exasollog` supports the following log levels:
+
+<dl>
+<dt>NONE</dt><dd>no log entries are generated on this level.</dd>
+<dt>FATAL</dt><dd>should be used for in cases of errors that require immediate program termination.</dd>
+<dt>ERROR</dt><dd>is best used in cases where the calling code should be given a chance to deal with the error or at least clean up before termination.</dd>
+<dt>WARN</dt><dd>stands for situations that don't have an immediate adverse effect, but might cause an error later.</dd>
+<dt>INFO</dt><dd>is for sparse log information that you want to appear in a standard log. This is also the default log level.</dd>
+<dt>CONFIG</dt><dd>should be used to log parameters and environment of the client.</dd>
+<dt>DEBUG</dt><dd>is targeted at developers when trying to analyze problems.</dd>
+<dt>TRACE</dt><dd>is showing details of internal state.</dd>
+</dl>
+
+You can change the log level with the function `set_level()`:
+
+```lua
+log.set_level("DEBUG")
+```
+
+### Writing log Messages
+
+Writing log messages is as easy as it gets:
+
+```lua
+log.debug("Initialization complete.")
```

Comment:
```suggestion
log.info("Initialization complete.")
```

debug is hidden by default

Reply:
This is intentionally on level `DEBUG` to not clutter the log.

Reply:
Yes but as an example here, I first tried it out and revcieved nothing... That's disapointing